packs/BP/scripts/network_ipc.ts

```diff
@@ -106,19 +106,24 @@ export function generateListener(payload: ipc.SerializableValue): null {
   const location = deserializeDimensionLocation(data.a);
   const type = data.b;
   const amount = data.c;
+  const useReserveStorage = data.d;
+  const consumeExisting = data.e;
```

Comment:
`consumeExisting` should be removed or made into a different function, as it doesn't make sense here. If `useReserveStorage` and `consumeExisting` are both `true`, then `amount` (a required argument) is never used.

Reply:
issue resolved by using an enum called mode instead of indivudal parameters to control each thing